modules/stochastic_tools/src/utils/BootstrapCalculators.C

```diff
@@ -56,6 +59,25 @@ BootstrapCalculator::BootstrapCalculator(const libMesh::ParallelObject & other,
               "The supplied levels must be less than one");
 }
 
+std::vector<Real>
+BootstrapCalculator::computeBootstrapEstimates(const std::vector<Real> & data,
+                                               const Calculator & calc,
+                                               const bool is_distributed) const
+{
+  MooseRandom generator;
+  generator.seed(0, _seed);
+
+  // Compute replicate statistics
+  std::vector<Real> values(_replicates);
+  for (std::size_t i = 0; i < _replicates; ++i)
+  {
+    std::vector<Real> replicate = shuffle(data, generator, is_distributed);
+    values[i] = calc.compute(replicate, is_distributed);
```

Comment:
Is this what you and Cody were talking about in regards to a bunch of small communications?

Reply:
No, the small communications revolve around the acceleration calculation. But, this is also an area that could be a problem as well. Although, the number of calculations here is only on the order of 1000 to 10,000.